### PR TITLE
DEV-5632 navigation sidebar fixes

### PR DESCRIPTION
**Pull request by jameslee40**
**High level description:**
Adds option in `Sidebar` for `Data Sources and Methodology`.
Adds `SidebarFooter` component underneath sidebar which contains a link to `Data Lab`.
Fixes inconsistencies with active `Sidebar` section when scrolling if some table rows are expanded.
Fixes bug when clicking on `Overview` and then subsequently another tab, would add too much spacing on the top.

**JIRA Ticket:**
[DEV-5632](https://federal-spending-transparency.atlassian.net/browse/DEV-5632)

**Mockup:**
https://bahdigital.invisionapp.com/share/6NIAEAQCAM9#/screens

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [N/A] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [N/A] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [N/A] Added Unit Tests for methods in Container Components, reducers, helper functions, and models `if applicable`
- [N/A] All `componentWillReceiveProps` and `componentWillMount` in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3277](https://federal-spending-transparency.atlassian.net/browse/DEV-3277)
- [N/A] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [N/A] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [x] Design review complete `if applicable`
- [x] Code review complete

### CODE REVIEW

File: src/js/containers/covid19/Covid19Container.jsx [hunk 1]
```diff
@@ -174,25 +175,33 @@ const Covid19Container = () => {
...
-                            active={activeSection}
-                            jumpToSection={jumpToCovid19Section}
-                            detectActiveSection={setActiveSection}
-                            sections={Object.keys(componentByCovid19Section())
-                                .filter((section) => componentByCovid19Section()[section].showInMenu)
-                                .map((section) => ({
-                                    section: snakeCase(section),
-                                    label: componentByCovid19Section()[section].title
-                                }))} />
+                    <div className="sidebar__wrapper">
```
Comment (ebdabbs) [inline src/js/containers/covid19/Covid19Container.jsx:178]: I would swap these class names to `sidebar` wrapping something like `sidebar__content` to adhere to BEM. (`sidebar` as the block and `sidebar__content` as an element of that block)

File: src/js/components/covid19/SidebarFooter.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    return (
+        <div>
+            <div className={`${props.pageName}-sidebar-footer-reference ${floatSidebarFooter}`} ref={referenceDiv}>
+                &nbsp;
+            </div>
+            <div ref={div} className={`${props.pageName}-sidebar-footer ${floatSidebarFooter}`} style={{ width: sidebarFooterWidth, top: sidebarFooterTop }}>
+                <div className={`${props.pageName}-sidebar-footer__header`}>Learn more about the COVID-19 response</div>
+                <div className={`${props.pageName}-sidebar-footer__content`}>To learn more about COVID-19 spending legislation and the broader projected economic and fiscal impact of these dollars,
+                    <a href="https://datalab.usaspending.gov/"> visit our DATA Lab &nbsp;
```
Comment (ebdabbs) [inline src/js/components/covid19/SidebarFooter.jsx:67]: Verify with design - I think this is intended to link to to Data Lab's COVID page https://datalab.usaspending.gov/federal-covid-spending (not yet released)

File: src/js/containers/covid19/Covid19Container.jsx [hunk 3]
```diff
@@ -174,25 +175,33 @@ const Covid19Container = () => {
...
+                                pageName="covid19"
+                                fixedStickyBreakpoint={scrollPositionOfSiteHeader(Cookies.get('usaspending_covid_homepage'))} />
+                        </div>
                     </div>
                     <div className="body usda__flex-col">
                         <section className="body__section">
                             <Heading />
                         </section>
                         {Object.keys(componentByCovid19Section())
+                            .filter((section) => componentByCovid19Section()[section].showInMainSection)
```
Comment (ebdabbs) [inline src/js/containers/covid19/Covid19Container.jsx:204]: Nice!

File: src/js/containers/covid19/Covid19Container.jsx [hunk 4]
```diff
@@ -175,7 +175,7 @@ const Covid19Container = () => {
             </StickyHeader>
             <LoadingWrapper isLoading={isLoading}>
                 <main id="main-content" className="main-content usda__flex-row">
-                    <div className="sidebar__wrapper">
+                    <div className="sidebar__content">
```
Comment (ebdabbs) [inline src/js/containers/covid19/Covid19Container.jsx:178]:
```suggestion
                    <div className="sidebar">
```

File: src/js/containers/covid19/Covid19Container.jsx [hunk 5]
```diff
@@ -175,7 +175,7 @@ const Covid19Container = () => {
...
                         <div className="sidebar">
```
Comment (ebdabbs) [inline src/js/containers/covid19/Covid19Container.jsx:179]:
```suggestion
                        <div className="sidebar__content">
```

File: src/js/containers/covid19/Covid19Container.jsx (hunk 4 above)
Comment (ebdabbs) [reply src/js/containers/covid19/Covid19Container.jsx:178]: Sorry my previous comment wasn't super clear - the block (`sidebar`) should wrap the element (`sidebar__content`) ([BEM docs here](http://getbem.com/introduction/))

**Comment by jameslee40** [comment]
> Screen Shot 2020-07-23 at 1 31 46 PM
> 
> Now I'm seeing some overlapping once I scroll down on the page. Let me know if you're able to replicate!

Actually, I'm not able to replicate this one.

**Comment by ebdabbs** [comment]
Re-ran npm start and it's not happening anymore! 🎉

**Pull request merged**